pkg/ring/lifecycler.go

```diff
@@ -52,6 +60,7 @@ type LifecyclerConfig struct {
 	NormaliseTokens  bool          `yaml:"normalise_tokens,omitempty"`
 	InfNames         []string      `yaml:"interface_names"`
 	FinalSleep       time.Duration `yaml:"final_sleep"`
+	TokenFileDir     string        `yaml:"token_file_dir,omitempty"`
```

Comment:
Why not `TokensFilePath`? We always create only 1 file (called `tokens`) inside the dir, which looks more natural to allow to configure the full filepath.

BTW, I would call it `TokensFile...` instead of `TokenFile...` since it contains all tokens.

Reply:
Do you mean the entire path including the file name?

Reply:
Yes. If it's going to be just 1 file, configuring the whole path looks just more clear to me.

Reply:
I feel leaving the file name to the user would lead to problems or confusion (if names are changed). By only taking the directory we can be consistent with the file name.

Reply:
Ok. What's about calling it `TokensFileDir`?

Reply:
Sure. Will change.

Reply:
With some discussions offline with @gouthamve, filepath wins